File: Menu_example_Java/app/src/main/java/com/example/menu_java/MainActivity.java
package com.example.menu_java;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        setSupportActionBar(findViewById(R.id.toolbar));//Indico donde esta el XML del toolbar
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.menu_main, menu);//Inflo o genero el menu
        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        String itemSelected = item.getTitle().toString() + " selected";
        Toast.makeText(this, itemSelected, Toast.LENGTH_SHORT).show();
        return super.onOptionsItemSelected(item);
    }
}

File: Menu_example_Kotlin/app/src/main/java/com/example/menu_kotlin/MainActivity.kt
package com.example.menu_kotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.widget.Toast

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setSupportActionBar(findViewById(R.id.toolbar))//Indico donde esta el XML del toolbar
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        getMenuInflater().inflate(R.menu.menu_main, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val itemSelected = item.title.toString() + " selected"
        Toast.makeText(this, itemSelected, Toast.LENGTH_SHORT).show()

        return super.onOptionsItemSelected(item)
    }
}